### PR TITLE
Add version 5.8 and fix sqlite+column_metadata mismatch.

### PR DESCRIPTION
**Pull request by danlipsa**

### CODE REVIEW

**Comment by danlipsa** [comment]
Fixes: https://github.com/spack/spack/issues/15448

**Comment by danlipsa** [comment]
@chuckatkins Please review

File: var/spack/repos/builtin/packages/sqlite/package.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ class Sqlite(AutotoolsPackage):
             '(unsafe for <3.26.0.0 due to Magellan).')
 
     variant('rtree', default=False, description='Build with Rtree module')
-    variant('column_metadata', default=False, description="Build with COLUMN_METADATA")
+    variant('column_metadata', default=True, description="Build with COLUMN_METADATA")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sqlite/package.py]: Is this change necessary?

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sqlite/package.py]: This fixes https://github.com/spack/spack/issues/15448

**Comment by danlipsa** [comment]
@chuckatkins ping

File: var/spack/repos/builtin/packages/sqlite/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/sqlite/package.py]: It seems like #15448 implies that paraview _requires_ `sqlite+column_metadata`. Simply changing the default doesn't ensure that someone won't override this.

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sqlite/package.py]: The error comes from qhelpgenerator which does require sqlite+column_metadata. The problem is that python only requires sqlite. spack does not know how to resolve this kind of conflicting specs. The only way I know how to fix this is to enable the most comprehensive package by default. In this case sqlite+column_metadata.

Comment (glennpj) [reply var/spack/repos/builtin/packages/sqlite/package.py]: Python depends on sqlite if `+sqlite` is in its specs. When `python+sqlite` is set the sqlite dependency can be met with either `sqlite+column_metadata` or `sqlite~column_metadata`.  Setting an explicit dependency for paraview should do the trick, as in #15248. However, I see the following comment in the paraview package:
`# depends_on('sqlite') # external version not supported`
but I am not really sure what that means in light of #15248 using an external sqlite successfully.

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sqlite/package.py]: Yes, that works, but that puts the burden on the user to know what to do when an error happens. That is not ideal, because most users will report this as a bug.

Building the largest package needed by default, as I propose here, solves this problem. This has been done in other instances such as https://github.com/spack/spack/pull/14795
I think it is reasonable solution until spack learn to merge conflicting specification from two branches of a dependency tree.

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sqlite/package.py]: > ``# depends_on('sqlite') # external version not supported``
> but I am not really sure what that means in light of #15248 using an external sqlite successfully.

I agree. I don't think that comment is valid anymore. I'll remove it.

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sqlite/package.py]: > ``# depends_on('sqlite') # external version not supported``

Means that ParaView is using an internal sqlite. Note that the error that ParaView sees comes from Qt help builder.

Comment (glennpj) [reply var/spack/repos/builtin/packages/sqlite/package.py]: Then the qt package needs the dependency spec. The following from qt/package.py
```
    depends_on("sqlite", when='+sql', type=('build', 'run'))                                                            
    depends_on("sqlite+column_metadata", when='+sql%intel', type=('build', 'run'))
```
could become
```
    depends_on("sqlitesqlite+column_metadata", when='+sql', type=('build', 'run')) 
```

Comment (glennpj) [reply var/spack/repos/builtin/packages/sqlite/package.py]: Note that I am not opposed to anything here but just trying to understand where the change should happen. I suspect that the sqlite package is used more then the qt package and could affect more packages and cause more rebuilds.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/sqlite/package.py]: It will definitely cause rebuilds (my entire Python stack), but there's no other way around the concretization issue until the new concretizer comes out.

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sqlite/package.py]: Indeed, I don't know of another good way to fix this.  As I report in my issue
https://github.com/spack/spack/issues/15448
ParaView needs qt and python3. In turn, qt needs sqlite+column_metadata while python needs sqlite~column_metadata.  I've seen spack reporting a concretization error in other instances. In this case, it just builds sqlite~column_metadata (maybe because of the order) and tries to use that.

Comment (glennpj) [reply var/spack/repos/builtin/packages/sqlite/package.py]: >while python needs sqlite~column_metadata.

I am not seeing that. 
```console
$ spack spec paraview+qt ^sqlite+column_metadata|grep sqlite
    ^sqlite+column_metadata
            ^python@3.7.4%gcc@9.1.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4~uuid+zlib arch=linux-centos7-broadwell
                ^sqlite@3.30.1%gcc@9.1.0+column_metadata+fts~functions~rtree arch=linux-centos7-broadwell
```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/sqlite/package.py]: This is a backtracking issue, see https://github.com/spack/spack/issues/15478#issuecomment-598819244

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sqlite/package.py]: That is weird. I get:
<pre>
[~/projects/spack (develop $%=)]$ spack spec paraview+qt | grep sqlite
            ^python@3.7.6%gcc@7.4.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4~uuid+zlib arch=linux-ubuntu18.04-broadwell
                ^sqlite@3.30.1%gcc@7.4.0~column_metadata+fts~functions~rtree arch=linux-ubuntu18.04-broadwell
[~/projects/spack (develop $%=)]$ git log -1
commit 38d04e29ebe481bcc6d344e669ffca4ebb79bb9f (HEAD -> develop, origin/develop, origin/HEAD)
Author: Hans Pabst <hfp@users.noreply.github.com>
Date:   Fri Mar 13 20:44:26 2020 +0100

    LIBXSMM 1.15 (#15482)
    
    * LIBXSMM 1.15
    
    * LIBXSMM: renamed development version according to the related branch on GitHub.
[~/projects/spack (develop $%=)]$ 

</pre>

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sqlite/package.py]: Oh, I did not see that you added the ^sqlite+column_metadata. Yes, that would work, but it requires the users to know to pass that. I think most users won't know to do that and they would report this as a bug.

Comment (glennpj) [reply var/spack/repos/builtin/packages/sqlite/package.py]: Yes, but that is how spack operates, allowing the spec to be overridden. It does not seem like a bug to me.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/sqlite/package.py]: It's a bug in the sense that the concretizer should be smart enough to figure this all out on its own, but it currently isn't.

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sqlite/package.py]: I see dependencies as a way to specify alternatives, for instance ^mpich versus ^openmpi, or maybe specific versions for dependencies.  I think resolving conflicting specs is beyond what a user could know (unless he has a lot of experience with spack), and indeed they could be done automatically.

**Comment by danlipsa** [comment]
@scottwittenburg Please review

**Comment by danlipsa** [comment]
@chuckatkins ping

**Comment by danlipsa** [comment]
@chuckatkins @scottwittenburg Ping?

**Comment by danlipsa** [comment]
Thanks @chuckatkins !
